source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -163,6 +163,9 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
     for (auto ctx : ssl_contexts) {
       if (verify_mode != SSL_VERIFY_NONE) {
         if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.tls_async_cert_validation")) {
+          // TODO(danzh) do not set SSL_VERIFY_NONE, instead, retain the state
+          // in ContextImpl and let boring SSL always call the verify callback
+          // which checks the state before doing actual verification.
```

Comment:
Perhaps, to explain the situation a bit more, it should say something like:

TODO(danzh) Envoy's use of SSL_VERIFY_NONE does not quite match the actual semantics as a client. As a client, SSL_VERIFY_NONE means to verify the certificate (which will fail without trust anchors), save the result in the session ticket, but otherwise continue with the handshake. Envoy actually wants it to accept all certificates, so instead configure a callback to do that. (As a server, configuring SSL_VERIFY_NONE is correct. SSL_VERIFY_NONE as a server won't request client certificates, which is what we want.)